src/elements/content-sidebar/activity-feed/activity-feed/ActivityThread.js

```diff
@@ -0,0 +1,108 @@
+// @flow
+import * as React from 'react';
+import getProp from 'lodash/get';
+import classNames from 'classnames';
+import { FormattedMessage } from 'react-intl';
+
+import Comment from '../comment';
+import ActivityCard from '../ActivityCard';
+import ActivityItem from './ActivityItem';
+import PlainButton from '../../../../components/plain-button';
+
+import type { GetAvatarUrlCallback, GetProfileUrlCallback } from '../../../common/flowTypes';
+import type { Translations } from '../../flowTypes';
+import type { SelectorItems, User } from '../../../../common/types/core';
+import type { Comment as CommentType } from '../../../../common/types/feed';
+
+import messages from './messages';
+import './ActivityThread.scss';
+
+type Props = {
+    children: React.Node,
+    currentUser?: User,
+    getAvatarUrl: GetAvatarUrlCallback,
+    getMentionWithQuery?: Function,
+    getUserProfileUrl?: GetProfileUrlCallback,
+    id: string,
+    mentionSelectorContacts?: SelectorItems<>,
+    onGetReplies?: (id: string) => void,
+    onReplyDelete?: Function,
+    onReplyEdit?: Function,
+    replies?: Array<CommentType>,
+    repliesTotal?: number,
+    translations?: Translations,
+};
+
+const ActivityThread = ({
+    children,
+    currentUser,
+    getAvatarUrl,
+    getMentionWithQuery,
+    getUserProfileUrl,
+    id,
+    mentionSelectorContacts,
+    onGetReplies,
+    onReplyDelete,
+    onReplyEdit,
+    replies,
+    repliesTotal = 0,
+    translations,
+}: Props) => {
+    const [repliesExpanded, setRepliesExpanded] = React.useState(false);
+
+    const toggleReplyMessage = repliesExpanded ? messages.hideReplies : messages.getMoreReplies;
+    const repliesToLoad = repliesTotal - 1;
+
+    const toggleReplies = () => {
+        !repliesExpanded && onGetReplies && onGetReplies(id);
+        setRepliesExpanded(previousState => !previousState);
+    };
+
+    const renderReply = (reply: CommentType) => (
+        <ActivityItem key={reply.type + reply.id} className="bcs-ActivityThread-reply" data-testid="reply">
+            <Comment
+                {...reply}
+                currentUser={currentUser}
+                getAvatarUrl={getAvatarUrl}
+                getMentionWithQuery={getMentionWithQuery}
+                getUserProfileUrl={getUserProfileUrl}
+                mentionSelectorContacts={mentionSelectorContacts}
+                onDelete={onReplyDelete}
+                onEdit={onReplyEdit}
+                permissions={{
+                    can_delete: getProp(reply.permissions, 'can_delete', false),
```

Comment:
same here, should we pass `can_reply` and `can_resolve` as well?

Reply:
`can_reply` will be consumed only in ActivityThread component - there is no reason to pass it to Comment / Reply. 
But `can_resolve` can be handy - thanks!